### PR TITLE
Fixes Farmbot Pathing

### LINKED ISSUE #15482: Farmbots Don't Pathfind to Trays
title.

### PR DESCRIPTION
**Pull request by Doxxmedearly**
Fixes #15482
Partially handles  #16748

Farmbots usually "stop working" due to them getting hung up on an impossible-to-reach target. They will constantly re-target it even after the target is dropped due to an invalid path. It used to happen with sinks. Now, with the new map, it happens with trays.

First problem was that it would want to tend to trays or soil it could not reach; usually, this was soil, as the path to the public garden is blocked off. This was fixed by making sure we only accept trays we can path to as a valid target.

Next issue was with how GETTING the path worked. It used `get_dir()` and `get_step()` to find the open space nearest the tray. This was fine on previous maps, as the trays were usually on the edge of hydro and/or in a single row. However, now that the trays are arranged in rows two trays thick, sometimes `get_step()` would pick another tray that's between the bot and the target, making the path invalid while telling the bot that it was valid.

Now pathfinding tries to find any accessible adjacent space, and then we pick a path to that. This stops the bot from getting hung up on the northmost trays if it's near the south of hydro. 

tl;dr bots stopped bc target path issue. Issue fixed.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

**Comment by Doxxmedearly** [comment]
!bugfix

**Review by SleepyGemmy — APPROVED** [review]
